src/cargo/sources/git/known_hosts.rs

```diff
@@ -316,42 +386,70 @@ fn check_ssh_known_hosts_loaded(
             }
             continue;
         }
-        if key_matches {
-            return Ok(());
+        match known_host.line_type {
+            KnownHostLineType::Key => {
+                if key_matches {
+                    return Ok(());
+                }
```

Comment:
I believe there is an issue here. This should not early return on a key match. Otherwise the known_hosts file becomes order-dependent, particularly if the revoked key is found later in a file. I think this should only return until after all entries have been processed.

Reply:
You're completely right. Nice catch!

I've fixed the issue and added a test for this case for good measure.